variables.tf

```
@@ -0,0 +1,24 @@
+variable "resource_group_name" {
+  type    = string
+  default = "example-resources"
+}
+
+variable "location" {
+  type    = string
+  default = "West Europe"
+}
+
+variable "storage_account_name" {
+  type    = string
+  default = "examplestoragemaksimens"
```

In variables.tf: Azure Storage Account names must be globally unique across all of Azure, and they must be between 3 and 24 characters in length and use numbers and lowercase letters only. The hardcoded default value `examplestoragemaksimens` is highly likely to be already in use, which will cause `terraform apply` to fail. It's recommended to remove the `default` and provide a unique name when you run Terraform, or use a mechanism to generate a unique name.